### PR TITLE
Spike

### PR DESCRIPTION
**Pull request by aeneasr**

### CODE REVIEW

File: cmd/tools.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// +build tools
+
+package cmd
+
+import (
```
Comment (ermik) [inline cmd/tools.go]: Go best practices advise to comment on `_` imports to inform the reader of their... importance.

File: driver/registry_memory.go [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+type postHooks []interface {
+	selfservice.HookLoginPostExecutor
+	selfservice.HookRegistrationPostExecutor
+}
+
+func (m *RegistryMemory) hooksPost(credentialsType identity.CredentialsType, configs []configuration.SelfServiceHook) postHooks {
+	var i postHooks
+
+	for _, hook := range configs {
+		switch hook.Run {
```
Comment (ermik) [inline driver/registry_memory.go]: Could these switch cases be further refactored into case-specific functions? A lot is happening in one switch.

File: identity/pool_memory.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/go-convenience/stringslice"
+
+	"github.com/ory/herodot"
+	"github.com/ory/x/pagination"
+)
+
+var _ Pool = new(PoolMemory)
```
Comment (ermik) [inline identity/pool_memory.go]: Is this an interface conformance check?

File: identity/pool_memory.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	defer p.RUnlock()
+
+	start, end := pagination.Index(limit, offset, len(p.is))
+	return p.is[start:end], nil
+}
+
+func (p *PoolMemory) Update(_ context.Context, i *Identity) (*Identity, error) {
+	p.RLock()
+	for k, ii := range p.is {
+		if ii.ID == i.ID {
```
Comment (ermik) [inline identity/pool_memory.go]: How about:
```go
if ii.ID != i.ID {
        continue
}
// code  currently inside the conditional
```

File: identity/pool_memory.go (hunk 3 above)
Comment (aeneasr) [reply identity/pool_memory.go]: Yes

File: driver/registry_memory.go (hunk 2 above)
Comment (aeneasr) [reply driver/registry_memory.go]: Yeah most likely

**Pull request merged**